### DISCUSSION TITLE
Python API for dp train

### DISCUSSION THREAD

**Discussion by ZhihaoXu0313** [deepmodeling/deepmd-kit#2931] [Q&A] [answered]
Hi,

I just want to know if there's any python api for the training of deep potential, which can replace the command "dp train input.json" in command line? Thanks for any help or response!

Best,
ZX

**Comment by njzjz** [comment, marked as answer]
This is equivalent:

```
from deepmd.entrypoints.main import main as dp

dp(["train", "input.json"])
```

If you don't care about the performance, using `subprocess` will be more safe.

**Reply by ZhihaoXu0313** [reply]
Thank you Jinzhe! I’m now using the subprocess and I am just curious if there’s any “elegant” way to do that! That’s helpful!
